gui/packages/ubuntupro/lib/pages/landscape/landscape_model.dart

```diff
@@ -0,0 +1,134 @@
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+import '/core/agent_api_client.dart';
+
+enum LandscapeConfigType { manual, file }
+enum FileError { notFound, tooLarge, empty, none }
+
+class LandscapeModel extends ChangeNotifier {
+  LandscapeModel(this.client);
+  final AgentApiClient client;
+
+  LandscapeConfigType _selected = LandscapeConfigType.manual;
+
+  String _path = '';
+
+  String _fqdn = '';
+  String accountName = 'standalone';
+  String key = '';
+
+  bool _fqdnError = false;
+  FileError _fileError = FileError.none;
+
+  bool get fqdnError => _fqdnError;
+  FileError get fileError => _fileError;
+
+  set fqdn(String value) {
+    if (value.isNotEmpty && !value.startsWith('https://')) {
+      value = 'https://$value';
+    }
+    _fqdn = value;
+    validateFQDN();
+  }
+
+  String get fqdn => _fqdn;
+
+  set selected(LandscapeConfigType value) {
+    _selected = value;
+    _fqdnError = false;
+    _fileError = FileError.none;
+    validConfig();
+    notifyListeners();
+  }
+
+  LandscapeConfigType get selected {
+    return _selected;
+  }
+
+  set path(String value) {
+    _path = value;
+    validatePath();
+  }
+
+  String get path => _path;
+
+  bool validateFQDN() {
+    _fqdnError = _fqdn.isEmpty || Uri.tryParse(_fqdn) == null;
```

Comment:
FQDN also shouldn't have a port number. This check could then be something like:
```suggestion
   final uri = Uri.tryParse(_fqdn)
    _fqdnError = _fqdn.isEmpty || uri == null || uri.hasPort;
```